src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || this.a + this.b <= this.c
+      || this.a + this.c <= this.b
+      || this.b + this.c <= this.a
+    ) {
+      throw new Error('This triangle is not mathematically possible...  :C  ');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const halfOfPerimeter = (this.a + this.b + this.c) / 2;
+    const triangleArea = Math.sqrt(halfOfPerimeter * (halfOfPerimeter - this.a)
+    * (halfOfPerimeter - this.b) * (halfOfPerimeter - this.c));
 
+    return Math.round(triangleArea * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
```

Comment:
totally agree for the comment about using an enum
that's a common practice to avoid strings as much as possible
```suggestion
  shape: Shape = Shape.Triangle;
```